### ISSUE TITLE
Subtle differences in table definition between nodes may cause errors during aggregation

### ISSUE THREAD

**Issue by nickitat** [ClickHouse/ClickHouse#42630]
``` sql
create table shard_0.t_different_dbs(a UInt32, b UInt32) engine = MergeTree order by a;

create table shard_1.t_different_dbs(a UInt8, b UInt8) engine = MergeTree order by a;

create table dist_t_different_dbs as shard_1.t_different_dbs engine = Distributed(test_cluster_two_shards_different_databases, '', t_different_dbs);

insert into shard_0.t_different_dbs select number % 1000, number % 1000 from numbers_mt(1e6);
insert into shard_1.t_different_dbs select number % 1000, number % 1000 from numbers_mt(1e6);

select a, count() from dist_t_different_dbs group by a order by a limit 5 offset 500 settings group_by_two_level_threshold = 1;
```

Returns error: `Code: 93. DB::Exception: Received from localhost:9000. DB::Exception: Unknown aggregated data variant.: While executing GroupingAggregatedTransform. (UNKNOWN_AGGREGATED_DATA_VARIANT)`.

The reason is HT of type `key8` is not convertible to two-level hash table. We do convertion [here](https://github.com/ClickHouse/ClickHouse/blob/master/src/Processors/Transforms/MergingAggregatedMemoryEfficientTransform.cpp#L292).

Another funny case is:

``` sql
create table shard_0.t_different_dbs(a UInt32, b UInt32) engine = MergeTree order by a;
create table shard_1.t_different_dbs(a UInt64, b UInt64) engine = MergeTree order by a;

select a, count() from dist_t_different_dbs group by a, b order by a, b limit 5 offset 500 settings group_by_two_level_threshold = 1;
```

Returns incorrect result because we calculate hashes [differently](https://github.com/ClickHouse/ClickHouse/blob/master/src/Common/HashTable/Hash.h) for keys of size <= 8 bytes and keys of size > 8 byte.

**Comment by alexey-milovidov** [comment]
This is still relevant.

**Comment by clickgapai** [comment]
_Found via ClickGap automated review. Please close or comment if this is incorrect or needs adjustment._

### TL;DR
**Subtle differences in table definition between nodes may cause errors during aggregation**

**ClickGap verified:** reproduced it with the script below on `master`; tested 5 version(s) — 5 affected, 0 clean (see Affects); confirmed it predates the oldest available release (pre-existing).

### Reproduction
```
CREATE DATABASE IF NOT EXISTS shard_0;
CREATE DATABASE IF NOT EXISTS shard_1;
CREATE TABLE shard_0.t_different_dbs(a UInt32, b UInt32) ENGINE = MergeTree ORDER BY a;
CREATE TABLE shard_1.t_different_dbs(a UInt8, b UInt8) ENGINE = MergeTree ORDER BY a;
CREATE TABLE default.dist_t_different_dbs AS shard_1.t_different_dbs
    ENGINE = Distributed(test_cluster_two_shards_different_databases, '', t_different_dbs);
INSERT INTO shard_0.t_different_dbs SELECT number % 1000, number % 1000 FROM numbers_mt(1e6);
INSERT INTO shard_1.t_different_dbs SELECT number % 1000, number % 1000 FROM numbers_mt(1e6);
SELECT a, count() FROM default.dist_t_different_dbs GROUP BY a ORDER BY a LIMIT 5 OFFSET 500
    SETTINGS group_by_two_level_threshold = 1;
```

### Bisect
- Bisect skipped — bug predates available CI binaries. Maintainer-side bisect needed.

### Affects
- **Reproduces on**: `master, 26.5, 26.4, 26.3, 25.8`

### Suggested next step
Manual triage is needed. The reproducer above should be a fast first check.

**Severity:** `P2`

---
_ClickGap · Finding: `h_issue_42630`_

**Comment by nickitat** [comment]
The committee of experts decided not to fix it.
